app/models/ext_management_system.rb

```diff
@@ -173,6 +174,43 @@ def hostname_format_valid?
 
   default_value_for :enabled, true
 
+  after_save :change_maintenance_for_child_managers, :if => proc { |ems| ems.changed_attributes.include?('enabled') }
+
+  def disable!
+    _log.info("Disabling EMS [#{name}] id [#{id}].")
+    update!(:enabled => false)
+    _log.info("Disabling EMS [#{name}] id [#{id}] successful.")
+  end
+
+  def enable!
+    _log.info("Enabling EMS [#{name}] id [#{id}].")
+    update!(:enabled => true)
+    _log.info("Enabling EMS [#{name}] id [#{id}] successful.")
+  end
+
+  # Move ems to maintenance zone and backup current one
+  # @param orig_zone [Zone] children original zones can be replaced in provider-specific callbacks
+  def pause!(orig_zone = nil)
```

Comment:
Can you explain a bit more when would you want to override the previous zone?
I see you pass in backup_zone when pausing the child EMSs, why not just keep it simple and have backup_zone always be that ems' previous zone?

Reply:
Agreed...this also simplifies my suggestion [here](https://github.com/ManageIQ/manageiq/pull/17452#pullrequestreview-135606810)

Reply:
@agrare @Fryguy there is problem with many `ensure_managers()` / `ensure_managers_zone_and_provider_region` methods, which are before_create / before_update callbacks.
See following scenario:
1. parent and child EMS in default zone
2. <calling pause!> (without param)
3. parent zone set to Maintenance, child zone **may** be set to Maintenance by `ensure_managers` callback
4. child calling pause!
5. ==> child backup zone set to Maintenance :-1:

So why this approach:
`ensure_managers` are not only in core, it needs changes on more places and more PRs. 
Developer in the future implementing new provider with different `ensure_managers` method must keep in mind something about backup_zone (which (s)he is not => future bug)

Reply:
> parent zone set to Maintenance, child zone may be set to Maintenance by ensure_managers callback

So this is where I think things break down.  If an EMS has child managers and no way to keep their zone in sync that is a bug that we should fix, not workaround for a specific zone.

If we only pause the parent manager we could let the mechanism that normally keeps the zone in sync sort out the children.

Reply:
@agrare I like more the approach that this feature is compact and error proof and doesn't depend on some other functionality if it's not necessary. I don't think it's workaround, but behaviour specific for pausing providers.
But..I cannot merge :) so let's move to your solution, which will work too